cooked-validators/tests/Cooked/MockChain/Monad/StagedSpec.hs

```diff
@@ -72,66 +61,6 @@ tests =
                 validateTxSkel $ f $ txSkel [paysPK (walletPKHash $ wallet 2) (Pl.lovelaceValueOf 4200000)]
                 validateTxSkel $ g $ txSkel [paysPK (walletPKHash $ wallet 3) (Pl.lovelaceValueOf 4200000)]
                 validateTxSkel $ h $ txSkel [paysPK (walletPKHash $ wallet 4) (Pl.lovelaceValueOf 4200000)]
-           in interpret (somewhere f (tr id id id)) `imcEq` interpret (tr id (fromJust . f) id)
-      ],
-    testCase
-      "Somewhere is exponential in branch number"
-      $
-      -- If we make a trace @tr = a >> b@ with two transactions, Then execute:
-      --
-      -- > tr' = somewhere f $ somewhere g $ tr
-      --
-      -- And we expect there to be four traces as a result:
-      --
-      -- > 1. f (g a) >> b
-      -- > 2. f a >> g b
-      -- > 3. g a >> f b
-      -- > 4. a >> f (g b)
-      --
-      -- If we execute @somewhere k tr'@ instead, we should expect to see 8 branches.
-      -- Because we're choosing @f = g = k = id@, we exept the eight traces to be equal to tr.
-      let tr =
-            validateTxConstr [paysPK (walletPKHash $ wallet 2) (Pl.lovelaceValueOf 4_200_000)]
-              >> validateTxConstr [paysPK (walletPKHash $ wallet 2) (Pl.lovelaceValueOf 8_400_000)]
-       in interpret (somewhere Just $ somewhere Just $ somewhere Just tr)
-            `imcEq` asum (replicate 8 $ interpret tr),
-    testCase "Modality order is respected" $
```

Comment:
*: And this

Reply:
I'm not sure if this ensures a behaviour that one should expect in the framework we're looking at here. Since "after == and", do we really want to say that one should be allowed to rely on the fact that `f LtlAnd g` applies `g` first? Furthermore, since composition of modifications is user-definable, if you define it "the other way round", this behaviour will change. (This also  has something to do with the comment you made on the definition of `nowLater`: I just chose an arbitrary order there). 

I think it's sensible to test something like this:
```haskell
everywhere f (a >> everywhere g b >> c) == f a >> f (g b) >> f c
```
(in the case where `f . g == g . f`)

Reply:
It seems I opened a little can of worms with that example: As it stands at the moment, we should have
```haskell
everywhere f (a >> everywhere g b >> c) == f a >> f (g b) >> c
```
That's because `everywhere` uses `StopLtl`, which ends _all_ modifications. My suggestion would be to remove `StopLtl` completely; I had only added it as an afterthought to keep the old semantics of `somewhere` and `everywhere`. These would have to change, so that both modify the computation "from now on". As I already wrote in the discussion of issue #112, if we have "next", that should probably be the semantics of "apply an LTL formula" anyway.

Reply:
As far as I understand, `after` is `and` only when the atoms of LTL are unit -- i.e., we're only caring for satisfiability. When caring for modifications, `after` is similar to `and`, but non-commutative. 

> I think it's sensible to test something like this:
> ```everywhere f (a >> everywhere g b >> c) == f a >> f (g b) >> f c```

I think that that is the behavior I'd expect; `everywhere g b` takes "precedence" and modifies `b` with `g` before `f`.